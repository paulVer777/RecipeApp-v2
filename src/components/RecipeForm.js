import React from 'react'

export default class RecipeForm extends React.Component {

    constructor(props){
        super(props)
        this.state = {
            title: props.data ? props.data.title : '',
            preparationTime:props.data ? props.data.preparationTime : '',
            instructions:props.data ? props.data.instructions : '',
            difficulty:props.data ? props.data.difficulty : '',
        }
    }

    onEventHandler = (event) => {
        const data = event.target.value
        const propertyName = event.target.name
        
        this.setState((prevState) => ({[propertyName]:data}),()=> 

            this.props.submitter({
            title:this.state.title,
            preparationTime:this.state.preparationTime,
            instructions:this.state.instructions,
            difficulty:this.state.difficulty,
        })
        
    )}

    render(){
        return (
            <div className = 'recipe-form' >
                <form onSubmit = {this.onSubmitHandler}>
                    <input type = 'text' placeholder = 'title' name = 'title' value = {this.state.title} onChange = {this.onEventHandler} />
                    <input type = 'text' placeholder = 'preparation time' name = 'preparationTime' value = {this.state.preparationTime} onChange = {this.onEventHandler}/>
                    <textarea type = 'text' placeholder = 'instructions' name = 'instructions' value = {this.state.instructions} onChange = {this.onEventHandler}/>
                    <select name = 'difficulty' value = {this.state.difficulty} onChange = {this.onEventHandler}  >
                        <option>Easy</option>
                        <option>Medium</option>
                        <option>Hard</option>
                    </select>
                </form>
            </div>
        )
    }
}