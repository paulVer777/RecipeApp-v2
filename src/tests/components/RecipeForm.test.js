import React from 'react'
import { shallow } from 'enzyme'
import RecipeForm from '../../components/RecipeForm';
import {recipes2} from '../fixtures/recipes'

test('Should render recipe form correctly', () => {

    const wrapper = shallow(<RecipeForm />)
    expect(wrapper).toMatchSnapshot()
})

test('Should render recipe form with passed data', () => {

    const wrapper = shallow(<RecipeForm data = {recipes2[0]} />)
    expect(wrapper).toMatchSnapshot()

})

test('should set title to state', () => {

    const submitter = jest.fn()

    const wrapper = shallow(<RecipeForm submitter = {submitter} />)
    wrapper.find('input').at(0).simulate('change', {
        target:{
            value:'pizza',
            name:'title'
        }
    })
    expect(wrapper.state('title')).toBe('pizza')
    expect(submitter).toHaveBeenLastCalledWith({"difficulty": "easy", "instructions": "", "preparationTime": "", "title": "pizza"})
})