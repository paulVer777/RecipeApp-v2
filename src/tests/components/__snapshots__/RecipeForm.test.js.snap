// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render recipe form correctly 1`] = `
<div
  className="recipe-form"
>
  <form>
    <input
      name="title"
      onChange={[Function]}
      placeholder="title"
      type="text"
      value=""
    />
    <input
      name="preparationTime"
      onChange={[Function]}
      placeholder="preparation time"
      type="number"
      value=""
    />
    <textarea
      name="instructions"
      onChange={[Function]}
      placeholder="instructions"
      type="text"
      value=""
    />
    <select
      name="difficulty"
      onChange={[Function]}
      value="easy"
    >
      <option
        value="easy"
      >
        Easy
      </option>
      <option
        value="medium"
      >
        Medium
      </option>
      <option
        value="hard"
      >
        Hard
      </option>
    </select>
  </form>
</div>
`;

exports[`Should render recipe form with passed data 1`] = `
<div
  className="recipe-form"
>
  <form>
    <input
      name="title"
      onChange={[Function]}
      placeholder="title"
      type="text"
      value="chili"
    />
    <input
      name="preparationTime"
      onChange={[Function]}
      placeholder="preparation time"
      type="number"
      value={20}
    />
    <textarea
      name="instructions"
      onChange={[Function]}
      placeholder="instructions"
      type="text"
      value="asdasdazxczxczxczczxczxczczxczczxczxx"
    />
    <select
      name="difficulty"
      onChange={[Function]}
      value="medium"
    >
      <option
        value="easy"
      >
        Easy
      </option>
      <option
        value="medium"
      >
        Medium
      </option>
      <option
        value="hard"
      >
        Hard
      </option>
    </select>
  </form>
</div>
`;
